mongo/instance_store.go

```diff
@@ -50,9 +50,6 @@ func (m *Mongo) GetInstances(ctx context.Context, states []string, datasets []st
 	results := []*models.Instance{}
 	totalCount, err := QueryPage(ctx, f, offset, limit, &results)
 	if err != nil {
-		if mongodriver.IsErrNoDocumentFound(err) {
-			return results, 0, errs.ErrDatasetNotFound
-		}
```

Comment:
Is this intentional? It looks like `QueryPage` can actually return an `ErrNoDocumentFound`

Reply:
No QueryPage will not return a ErrNoDocumentFound, and so this check is actually unnecessary. I will remove it